draft-ietf-tls-dtls13.md

```diff
@@ -487,6 +487,53 @@ current epoch implementations SHOULD use the most recent past epoch
 which has matching bits, and then reconstruct the sequence number as
 described above.
 
+### Sequence Number Encryption {#sne}
+
+In DTLS 1.3, when records are encrypted, record sequence numbers are
+also encrypted. The basic pattern is that the underlying encryption
+algorithm used with the AEAD algorithm is used to generate a mask
+which is then XORed with the sequence number.
+
+When the AEAD is based on AES, then the Mask is generated by
+computing AES-ECB on the first 16 bytes of the ciphertext:
+
+~~~~
+  Mask = AES-ECB(sn_key, Ciphertext[0..15])
+~~~~
+
+When the AEAD is based on ChaCha20, then the mask is generated
+by treating the first 12 bytes of the ciphertext as the
+Nonce and the next 4 bytes as the nonce:
+
+~~~~
+  Mask = ChaCha20(sn_key, Ciphertext[0..12], Ciphertext[13..15])
+~~~~
+
+The sn_key is computed as follows: 
+
+~~~~
+   [sender]_sn_key  = HKDF-Expand-Label(Secret, "sn" , "", key_length)
+~~~~
+
+[sender] denotes the sending side. The Secret value to be used is described 
+in Section 7.3 of {{I-D.ietf-tls-tls13}}.
+
+The encrypted sequence number is computed by XORing the leading
+bytes of the Mask with the sequence number. Decryption is
+accomplished by the same process.
+
+In some (rare) cases there the ciphertext may be less than 16 bytes.
+This cannot happen with most of the DTLS AEAD algorithms because
+the authentication tag itself is 16 bytes, however some algorithms
+such as TLS_AES_128_CCM_8_SHA256 have a shorter authentication tag,
+and in combination with a short plaintext, the result might be
+less than 16 bytes. In this case, implementations MUST pad the
+plaintext out in order to make a suitable-length ciphertext.
```

Comment:
Perhaps an example would be useful?

Reply:
And I assume one pads with zeros?

Reply:
this is just with the conventional padding mechanism.